src/datreant/core/metadata.py

```diff
@@ -1,63 +1,512 @@
 
+    @property
+    def _read(self):
+        self._statefile = JSONFile(os.path.join(self._tree._treantdir,
+                                                self._statefilename),
+                                   init_state=self._init_state)
 
-class AggTreeLimb(with_metaclass(_AggTreeLimbmeta, object)):
-    """Core functionality for limbs attached to a View.
+        return self._statefile.read()
+
+
+@functools.total_ordering
+class Tags(Metadata):
+    """Interface to tags.
 
     """
-    _name = 'aggtreelimb'
+    _statefilename = 'tags.json'
 
-    def __init__(self, collection):
-        self._collection = collection
+    @staticmethod
+    def _init_state(jsonfile):
+        jsonfile._state = []
+
+    def __repr__(self):
+        return "<Tags({})>".format(self._list())
+
+    def __str__(self):
+        tags = self._list()
+        agg = "Tags"
+        majsep = "="
+        seplength = len(agg)
+
+        if not tags:
+            out = "No Tags"
+        else:
+            out = agg + '\n'
+            out = out + majsep * seplength + '\n'
+            for i in range(len(tags)):
+                out = out + "'{}'\n".format(tags[i])
+        return out
+
+    def __getitem__(self, value):
+        # check if we might have a string to parse into a selection object
+        if isinstance(value, string_types):
+            value = parse_selection(value)
+        return self._getselection(value)
+
+    def _getselection(self, value):
+        """get tags according to a selection
+        """
+        with self._read:
+            if isinstance(value, list):
+                # a list of tags gives only members with ALL the tags
+                fits = all(self._getselection(item) for item in value)
+            elif isinstance(value, tuple):
+                # a tuple of tags gives members with ANY of the tags
+                fits = any(self._getselection(item) for item in value)
+            if isinstance(value, set):
+                # a set of tags gives only members WITHOUT ALL the tags
+                # can be used for `not`, basically
+                fits = not all(self._getselection(item) for item in value)
+            elif isinstance(value, string_types):
+                fits = value in self
+
+            return fits
+
+    def __iter__(self):
+        return self._list().__iter__()
+
+    def __len__(self):
+        return len(self._list())
+
+    def __eq__(self, other):
+        if isinstance(other, (Tags, set, list)):
+            return set(self) == set(other)
+        else:
+            raise TypeError("Operands must be tags, a set, or list.")
+
+    def __lt__(self, other):
+        if isinstance(other, (Tags, set, list)):
+            return set(self) < set(other)
+        else:
+            raise TypeError("Operands must be tags, a set, or list.")
+
+    def __sub__(self, other):
+        """Return a set giving the Tags in `a` that are not in `b`.
+
+        """
+        from .metadata import AggTags
```

Comment:
This line is cool. You are saying import class from this file.

Reply:
Ha...that's an artifact from before when these classes where in a separate file. I can probably get rid of it.